packages/core-users/src/module/configureUsersModule.ts

```diff
@@ -374,5 +373,41 @@ export const configureUsersModule = async ({
         user: removeConfidentialServiceHashes(user),
       });
     },
+
+    addPushSubscription: async (userId, subscription, { userAgent, unsubscribeFromOtherUsers }) => {
+      if (unsubscribeFromOtherUsers) {
```

Comment:
@Mikearaya I'd do it this way:

```
result = updateOne(_id equals logged in userId & pushSubscriptions.keys.p256dh != new push subscription key)
if (result shows updateOne found and modified a user && unsubscribeFromOtherUsers) {
   // unsubscribe all other users
}
```

Reply:
Because now what you do with your change is you added another roundtrip to the db with the findOne and still unsubscribe other users even when the $push was not successful

Reply:
And now it's another logic, it only adds the push subscription if not other user has the same key which is not what we want. I want to more or less keep the logic from before but optimize it for eventual consistency of mongodb.